### ISSUE TITLE
LightClient::get_or_fetch_block implicitly assumes that blocks of certain height which are `Failed` could now be different

### ISSUE THREAD

**Issue by OStevan** [informalsystems/tendermint-rs#435]
@milosevic @liamsi @romac [Ignoring](https://github.com/informalsystems/tendermint-rs/blob/master/light-client/src/light_client.rs#L249) `Failed` headers when trying to find a header of a certain height assumes that when requested again the light block will now be different for the same height. For me this is somewhat confusing that a correct node is able to change light blocks for the same height between two requests and it would seem wrong that we would treat this new header `Unverified` instead of immediately `Failed` since as far as I understand this is not something we would expect from a correct peer.

**Comment by liamsi** [comment]
> when requested again the light block will now be different for the same height. For me this is somewhat confusing that a correct node is able to change light blocks for the same height between two requests 

Yeah, sounds like this is a condition where the light-node should remove the peer and maybe gossip (?) this in the network too?
